### ISSUE TITLE
Move endListeningForKeyboard to viewWillDisappear

### ISSUE THREAD

**Issue by piercifani** [jessesquires/JSQMessagesViewController#1000]
Right now, when the `JSQMessagesComposerTextView` is the firstResponder, the animation when the user taps the Back button is anything but smooth, since it's bot animating the `JSQMessagesCollectionView` down to dismiss the keyboard and to the whole viewController's view to the right to make the pop animation of `UINavigationController`. [Example](http://www.gfycat.com/HarshIdioticAffenpinscher).

One way of fixing this is changing where the `endListeningForKeyboard` is called:

```
- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    [self jsq_addActionToInteractivePopGestureRecognizer:NO];
    self.collectionView.collectionViewLayout.springinessEnabled = NO;
    [self.keyboardController endListeningForKeyboard];
}

- (void)viewDidDisappear:(BOOL)animated
{
    [super viewDidDisappear:animated];
    [self jsq_removeObservers];
}
```

This will make it [definitely prettier](http://www.gfycat.com/ThinUniformBlackcrappie)

Will this change be breaking anything? Can I file a PR with this?

EDIT: Issue 1000!!! 🎉

**Comment by jessesquires** [comment]
thanks @piercifani 

however, i can't reproduce this on develop

**Comment by piercifani** [comment]
I could easily reproduce this on latest develop's demo project. [Here is a GIF](http://www.gfycat.com/HeartfeltBriefDiamondbackrattlesnake) showing it.

**Comment by jessesquires** [comment]
Hm... are you sure this is the latest? Something is off. I remember this issue, but it's been fixed for awhile.

**Comment by jessesquires** [comment]
Oh I see now. I was dragging back. This happens when "< Back" is tapped. :+1:

**Comment by jessesquires** [comment]
My current suspicion is that moving `endListeningForKeyboard` is going to have unintended side effects

**Comment by piercifani** [comment]
Exactly my concern and why I opened an issue instead of just a PR with it. Is this something that's worth investigating?

**Comment by cnowak7** [comment]
I don't know if this will help, but when I want to dismiss the keyboard when my chat view is scrolling to another view, I just do this:

```
- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    [self.view endEditing:YES];
}
```

Works pretty fine for me.

**Comment by jessesquires** [comment]
Fixed by 752fd38aa8b899cbe2e21b6595de58920977bfc2
